kura/org.eclipse.kura.net.admin/src/main/java/org/eclipse/kura/net/admin/NetworkAdminServiceImpl.java

```diff
@@ -330,147 +330,144 @@ public void updateEthernetInterfaceConfig(String interfaceName, boolean autoConn
-                                        newNetConfigs.add(netConfig);
-                                        configurationChanged = true;
-                                        if (!modifiedInterfaceNames.contains(interfaceName)) {
-                                            modifiedInterfaceNames.add(interfaceName);
-                                        }
                                     }
-                                    if (netConfig instanceof FirewallAutoNatConfig && !hadNatConfig) {
-                                        logger.debug("adding new FirewallAutoNatConfig to existing config for {}",
-                                                interfaceName);
-                                        newNetConfigs.add(netConfig);
+                                }
+                            } else if (netConfig instanceof FirewallAutoNatConfig) {
+                                if (natConfig == null) {
+                                    logger.debug("removing FirewallAutoNatConfig for {}", interfaceName);
+                                    configurationChanged = true;
+                                    if (!modifiedInterfaceNames.contains(interfaceName)) {
+                                        modifiedInterfaceNames.add(interfaceName);
+                                    }
+                                } else {
+                                    hadNatConfig = true;
+                                    newNetConfigs.add(natConfig);
+                                    if (!netConfig.equals(natConfig)) {
+                                        logger.debug("updating FirewallAutoNatConfig for {}", interfaceName);
                                         configurationChanged = true;
                                         if (!modifiedInterfaceNames.contains(interfaceName)) {
                                             modifiedInterfaceNames.add(interfaceName);
                                         }
+                                    } else {
+                                        logger.debug(
+                                                "not updating FirewallAutoNatConfig for {} because it is unchanged",
+                                                interfaceName);
                                     }
                                 }
+                            } else {
+                                logger.debug("Found unsupported configuration: {}", netConfig.toString());
                             }
+                        }
 
-                            for (NetConfig netConfig : newNetConfigs) {
-                                logger.debug("New NetConfig: {} :: {}", netConfig.getClass().toString(),
-                                        netConfig.toString());
+                        // add configs that did not match any in the current configuration
+                        if (netConfigs != null && !netConfigs.isEmpty()) {
```

Comment:
Is it needed to have this `!netConfigs.isEmpty()`?
Isn't that covered by the inner for loop?